mlir/include/mlir/Dialect/Linalg/Utils/Utils.h

```diff
@@ -38,6 +38,12 @@ namespace linalg {
 // General utilities
 //===----------------------------------------------------------------------===//
 
+// Returns true if `val` represents a zero-filled tensor, per its defining op.
+bool isZeroTensor(Value val);
+
+// Returns true if the operation defines a zero-filled tensor.
+bool isZeroOp(Operation *);
```

Comment:
[nit] I'd try a bit more descriptive name
```suggestion
bool isZeroLikeOp(Operation *);
```

Also, this is not really used at the Linalg level, is it? It's only needed in Utils.cpp, so I'd skip declaring it here.

Reply:
I've removed this.